playground/TestPlatform.Playground/Program.cs

```diff
@@ -49,42 +53,97 @@ static void Main(string[] args)
                 </RunSettings>
             ";
             var sources = new[] {
-                Path.Combine(playground, "MSTest1", "bin", "Debug", "net472", "MSTest1.dll")
+                (Path.Combine(playground, "MSTest1", "bin", "Debug", "net472", "MSTest1.dll"), new TestRunHandler("net472"), new DebuggerTestHostLauncher("net472")),
+                (Path.Combine(playground, "MSTest1", "bin", "Debug", "net48", "MSTest1.dll"), new TestRunHandler("net48"), new DebuggerTestHostLauncher("net48")),
             };
 
-            var options = new TestPlatformOptions();
-            r.RunTestsWithCustomTestHost(sources, sourceSettings, options, new TestRunHandler(), new DebuggerTestHostLauncher());
+            var tasks = new List<Task>();
+            foreach (var (source, handler, launcher) in sources)
+            {
+                var options = new TestPlatformOptions();
+                tasks.Add(Task.Run(() => r.RunTestsWithCustomTestHost(new[] { source }, sourceSettings, options, handler, launcher)));
+            }
+
+            await Task.WhenAll(tasks);
+
+            var tasks2 = new List<Task>();
+            foreach (var (source, handler, launcher) in sources)
+            {
+                var options = new TestPlatformOptions { };
+                tasks2.Add(Task.Run(() =>
+                {
+                    r.DiscoverTests(new[] { source }, sourceSettings, options, handler);
+                    var tests = handler.DiscoveredTests;
+                    if (!tests.Any())
+                    {
+                        // REview: the second discovery can get cancelled too early. This is task that will probably be solved by fixing the cancellation in the translation layer sender and how it receives messages.
```

Comment:
```suggestion
                        // Review: the second discovery can get cancelled too early. This is task that will probably be solved by fixing the cancellation in the translation layer sender and how it receives messages.
```

Reply:
Yeah, that task in the comment needs to be completed before merging.